src/Bicep.LangServer.UnitTests/Completions/PublicRegistryModuleMetadataProviderTests.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Bicep.LanguageServer.Providers;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Bicep.LangServer.UnitTests.Completions
+{
+    [TestClass]
+    public class PublicRegistryModuleMetadataProviderTests
+    {
+        private static PublicRegistryModuleMetadataProvider publicRegistryModuleMetadataProvider = new PublicRegistryModuleMetadataProvider();
+
+        [ClassInitialize]
+        public static async Task ClassInitialize(TestContext testContext)
+        {
+            await publicRegistryModuleMetadataProvider.Initialize();
```

Comment:
The `.UnitTests` & `.IntegrationTests` libraries should be able to run without any external service dependencies - do we need a way of injecting a mock for the live endpoint?

As a sanity check, try disabling your internet connection and verifying all the tests pass.

Reply:
+1 We don't want our CI to be broken if another service is having an outage.

Reply:
I deliberately added these tests. Since we own the endpoint(https://live-data.bicep.azure.com/modulesMetadata), I wanted to make sure we get notified when something is wrong. There shouldn't be any issues running these tests on CI. The only issue would be when these tests are run locally without internet connection.

Let me know if you guys think we don't need to worry about what gets returned from the live endpoint and I'll go ahead and update the tests to use mock.

Reply:
That's fair, definitely understand the motivation, and I agree on the value of having live tests.

It's unfortunately a bit of a gap in our testing framework capabilities. For CLI, we have "live tests", which operate against a real environment: https://github.com/Azure/bicep/tree/main/src/Bicep.Cli.E2eTests, but we don't have a similar capability for VSCode. The closest thing we have is the VSCode e2e tests (https://github.com/Azure/bicep/tree/main/src/vscode-bicep/src/test/e2e), but these don't currently assume access to a live environment.

For the sake of unblocking the PR, I'd recommend using mocks for the test library, and adding an "tech debt" issue that we can address later. There are a number of other features that I think would benefit from having a "live test" capability for VSCode (deploy Bicep, insert resource, probably others...)

Reply:
Thanks for the detailed explanation. 

I created below issue to add "live test"
https://github.com/Azure/bicep/issues/9963

And updated the tests to use mock here: https://github.com/Azure/bicep/commit/7fe067e6dc38c90b704199ce1d889da9e8263e85. I deleted the tests that I added for PublicRegistryModuleMetadataProvider.cs and included those scenarios in ModuleReferenceCompletionProviderTests.cs